tools/slurm/scripts/slurmcluster.sh

```diff
@@ -15,6 +15,29 @@ export OPT_LAUNCHER_PORT=8081
 MACHINE_TYPE=
 GPUS=
 
+tryloop_attempt() {
+    # Run the command after the `--` arg.
+    while [ "$1" != "--" ]; do shift; done
+    shift
+    "$@"
+}
+
+# Try an action multiple time with configurable sleeps in between.
+#
+# usage: tryloop A B... -- CMD...
+#
+# Equivalent to: CMD || sleep A && CMD || sleep B && CMD || ...
+tryloop() {
+    while [ "$1" != "--" ]; do
+        tryloop_attempt "$@" && return
+        echo "retrying in $1..." >&2
+        sleep "$1"
+        shift
+    done
+    # Final attempt.
+    tryloop_attempt "$@"
+}
```

Comment:
This is sick. Love this as an explicit exponential backoff config declaration.

Reply:
thanks!